### DISCUSSION TITLE
Location of xdrip wear apk

### DISCUSSION THREAD

**Discussion by Npavkov** [NightscoutFoundation/xDrip#3320] [Q&A]
Where is the wear apk for xdrip in the newest nightly build of xdrip?

**Comment by Navid200** [comment]
https://github.com/NightscoutFoundation/xDrip/discussions/3261

**Comment by Npavkov** [comment]
Navid, I cant seem to get apktool to work for me.....is there another way. can someone extract it and send it to me? I also can not get the "force update" option on the watch to work, it says "adb does not seem to be working"?

**Reply by Navid200** [reply]
That's not a good idea.  What will you do when the next release comes out?

**Comment by Npavkov** [comment]
can the wear component be provided as a separate apk download?

**Reply by Navid200** [reply]
It is my observation that there is no active developer who is supporting WearOS. 
I help with what I know.  I have no experience with WearOS.  So, I can do nothing to help.  

There are many discussions and requests about WearOS.  But, no one has stepped forward to support WearOS.

**Comment by Npavkov** [comment]
it is very sad that no one on GITHUB has abilities to continue development and support of xdrip wear os watch component. I feel that is the best asset of xdrip users to have their readings on their wrist.

**Reply by Navid200** [reply]
If the only advantage of WearOS is a matter of preference/style/convenience, I am not too worried I am sorry to say.

However, I understand there are parents who don't like to send their child to school with a phone because the child may easily lose the phone or forget to bring it back home.  So, they prefer to have a wrist watch the child can wear.
But, my limited understanding of how WearOS works is that it needs a phone to connect to the internet.  That will not address the issue because the child will still need to have a phone.

If someone can show me how WearOS can be used in this situation meaning a parent can be helped by allowing their child to go to school while they can follow the child with the child only having a WearOS watch on, I will go buy a WearOS watch and attempt to support it myself.

But, I don't believe WearOS is capable of doing that.

**Comment by PhilTU** [comment]
> how WearOS works is that it needs a phone to connect to the internet

Wi-Fi can be used directly by a watch, and GSM/LTE if the watch has that capability. https://developer.android.com/training/wearables/data/network-access

It's correct to say that network traffic goes via a phone when a phone is paired by Bluetooth and that phone has a network connection.

**Reply by Navid200** [reply]
Are parents using this then?
Do you know any parent who sends their kid to school with only a WearOS watch and they can follow the child as long as the child in away from home and at school and on the school bus?

I mean is this a normal watch or is it something a manufacturer created that no one in reality uses?

**Reply by PhilTU** [reply]
Parents like the convenience of a watch for the child (or themselves) as it allows the phone to be kept in a pocket or bag. I know of people who use this. It's much less of an issue for a child to look at a watch in class than get a phone out.

WearOS watches with GSM / LTE phone network connections are less common and more expensive. Wi-Fi is very common - Samsung GW4-6, Ticwatch Pro, Fossil Gen 5-6, Google Pixel Watch etc.

The options for direct connecting a watch (without a phone) to a sensor with xDrip+ are limited in any case, collector options at https://github.com/NightscoutFoundation/xDrip/blob/master/Documentation/WatchGuide.md#collector ,  so very few Libre users have it as an option. Following with xDrip+ sync or Nightscout follow might be a more common case for parents.

**Reply by Navid200** [reply]
Everything I say here is my personal view.

There are many different sections in the xDrip code.  They are not all created by the same person.

There are components that are crucial.  Think of the alerts.  It would be unimaginable if one day I come here and say "no one is supporting the alerts any longer".
Even though I have not written the alerts, but, I have spent a lot of time and continue to spend time on alerts so that I can address any issues if no one else does.
In my book, xDrip without alerts has no reason to exist.

I am going to repeat my comment in that context now.
If someone can show me that WearOS is a crucial part of xDrip, like the alerts are, I will learn the WearOS portion and support it myself.

If a parent has to send their child to school with a phone anyway, I wouldn't consider that a solution.

Dexcom is adding the ability to use a watch as a collector.  Let's wait for that and see if that may make any significant changes to the landscape.

Everything I state here is my personal view.

**Comment by PhilTU** [comment]
> If someone can show me that WearOS is a crucial part of xDrip

I won't attempt to do that. It's a useful part that some / many value. Currently it appears the updating of the existing WearOS watch component apk is broken due to changes in WearOS.
